ps_socialfollow.php

```diff
@@ -187,11 +202,32 @@ public function getConfigFieldsValues()
     {
         $result = [];
         foreach (static::SOCIAL_NETWORKS as $social) {
-            $result['blocksocial_' . $social] = Configuration::get('BLOCKSOCIAL_' . strtoupper($social));
+            if (!empty(Configuration::get('BLOCKSOCIAL_' . strtoupper($social)))) {
+                $this->upgradeConfiguration('BLOCKSOCIAL_' . strtoupper($social));
+            }
+            foreach ($this->context->controller->getLanguages() as $lang) {
+                $id_lang = $lang['id_lang'];
+                $conf = Configuration::get('BLOCKSOCIAL_' . strtoupper($social), $id_lang);
+                $result['blocksocial_' . $social][$id_lang] = $conf;
+            }
         }
         return $result;
     }
 
+    protected function upgradeConfiguration($name)
+    {
+        $conf = Configuration::get($name);
```

Comment:
```suggestion
        $configuration = Configuration::get($name);
```
When possible, I think it's better to avoid abbreviated variable names.

Full names avoid confusion (more details in https://softwareengineering.stackexchange.com/questions/46894/are-short-abbreviated-method-function-names-that-dont-use-full-words-bad-practi for example)

Maybe we can also add a little phpDoc that would explain what `$name` is, its type, and what happens inside this function ? 😄

Reply:
I prefer `$value` for configuration value, `$configuration` is long, and `$configuration_localized` is too long.

Reply:
Honest question: why "too long" ?

I love very long names 😄  like `$newConfigurationSubmittedByBOUser` but I used to be a Java developer and Java is known for very long names. I like to think "a long name will carry a lot of meaning and I make sure it's not confusing". But maybe I'm wrong so I'm curious: what is "too long" and why would it be an issue in your opinion?

Reply:
I think that's just a matter of personal preference. Factually too long would be, when the overly long name comes into conflict with the recommendation to keep lines short. Like PSR-2 `The soft limit on line length MUST be 120 characters; automated style checkers MUST warn but MUST NOT error at the soft limit. Lines SHOULD NOT be longer than 80 characters (...)`